rootfs/scheduler/__init__.py

```diff
@@ -993,6 +993,10 @@ def _wait_until_pods_are_ready(self, namespace, container, labels, desired):  #
             logger.info('timed out ({}s) waiting for pods to come up in namespace {}'.format(timeout, namespace))  # noqa
 
         logger.info("{} out of {} pods in namespace {} are in service".format(count, desired, namespace))  # noqa
+        if count != desired:
+            # raising to allow operations to rollback
+            raise KubeException('Not enough pods in namespace {} came into service. '
+                                '{} out of {}'.format(namespace, count, desired))
```

Comment:
we definitely should opt for failure mode when it is actually fails to starts.....both those bugs show the scenario of pods starting to fail or liveness probe failing which are failing modes...my question is regarding the readiness check where the pod actually starts running but isn't just ready to serve requests(this is definitely not ready but i don't think a failure) also we should have separate liveness check and readiness check instead of single health check
My point is it would be better if we have discussion on what is considered a failure mode though considering all scenarios as failure modes isn't gonna harm anything